phobos_umss.sh

```diff
@@ -0,0 +1,490 @@
+        STATUS="FAILURE"
+        save_log stageToCache "STATUS=$STATUS($error_2)"
+    fi
+    save_log stageToCache "END stageToCache($*)"
+    return $error_2
+}
+
+# function to create a new directory $1 in the MSS logical name space
+# phobos is not supposed to create subdirectories within its media, beacause it is an Object Store system.
+# Thus...
+# - In the case of a phobos instance managing POSIX directories, we can only see the creation of a new directory as the creation of a new media, which doesn't feature any link with its relatives.
+# - In the case of a phobos instance managing Tapes, apart from the very specific case of adding a new tape or driver to phobos scope, there is no sense in $(mkdir).
+mkdir () {
+    # <your command to make a directory in the MSS> $1
+    # e.g.: /usr/local/bin/rfmkdir -p rfioServerFoo:$1
+
+    save_log mkdir "BEGIN mkdir($*)"
+    op_mk=$(type -P mkdir)
+    op_ph=$(type -P phobos)
+    op_rd=$(type -P readlink)
+    
+    # Check that the directory is not yet registered in phobos.
+    check_exist_script="sudo $op_ph dir list ${1}"
+    check_exist=$($check_exist_script)
+    error_3=$?
+    save_log mkdir "$check_exist_script return with status=$check_exist($error_3)"
+    if [ "${check_exist}" != "${1}" ]; then
+        # Create a new directory in the file system
+        create_dir_script="sudo $op_mk -p ${1}"
+        create_dir=$($create_dir_script)
+        error_3=$?
+        save_log mkdir "$create_dir_script return with status=$create_dir($error_3)"
+        if [ $error_3 == 0 ]; then
+            # Phobos add this directory
+            full_path_script="sudo $op_rd -f ${1}"
+            full_path=$($full_path_script)
+            dir_add_script="sudo $op_ph dir add ${full_path}"
+            dir_add=$($dir_add_script >> $LOGFILE_FANCY 2>&1)
+            error_3=$?
+            save_log mkdir "$dir_add_script return with status=$dir_add($error_3)"
+            if [ $error_3 == 0 ]; then
+                # Phobos format this directory
+                dir_format_script="sudo $op_ph dir format ${full_path}"
+                dir_format=$($dir_format_script >> $LOGFILE_FANCY 2>&1)
+                error_3=$?
+                save_log mkdir "$dir_format_script return with status=$dir_format($error_3)"
+                if [ $error_3 == 0 ]; then
+                    # Phobos unlock this directory
+                    dir_unlock_script="sudo $op_ph dir unlock ${full_path}"
+                    dir_unlock=$($dir_unlock_script >> $LOGFILE_FANCY 2>&1)
+                    error_3=$?
+                    save_log mkdir "$dir_unlock_script return with status=$dir_unlock($error_3)"
+                fi
+            fi
+        fi
+    else
+        save_log mkdir "$1 already exists and is within phobos scope"
+    fi
+
+    if [ "$error_3" != 0 ]; then
+        STATUS="FAILURE"
+        save_log mkdir "STATUS=$STATUS($error_3)"
+    fi
+    save_log mkdir "END mkdir($*)"
+    return $error_3
+}
```

Comment:
major: as already discussed, there is no reason to link directories in a posix file tree and a DIR in phobos. The both are different concepts.

Reply:
Right.
In the next commit, the function performs no operation.